src/Microsoft.AspNet.Http.Extensions/SessionCollectionExtensions.cs

```diff
@@ -56,5 +56,15 @@ public static void Set(this ISessionCollection session, string key, byte[] value
         {
             session.Set(key, new ArraySegment<byte>(value));
         }
+        
+        public static T Get<T>(this ISessionCollection session, string key, ISessionFormatter<T> formatter)
+        {
+            return formatter.Deserialize(session.Get(key));
+        }
+
+        public static void Set<T>(this ISessionCollection session, string key, T value, ISessionFormatter<T> formatter)
+        {
+            session.Set(key, formatter.Serialize(value));
+        }
     }
 }
```

Comment:
When committing from Windows to GitHub: always a line-break at the end of each file, so the diff does not show this:
<img src="http://i.stack.imgur.com/IiQyS.jpg" alt="github: no newline before the end of line" />

Read more on [No newline before the end of file - Stackoverflow](http://stackoverflow.com/questions/5813311/no-newline-at-end-of-file).